CollapseLauncher/Classes/Helper/Database/DBHandler.cs

```diff
@@ -10,355 +11,412 @@
-            // The error should be handled in the method caller instead
-            catch (LibsqlException e) when (e.Message.Contains("`api error: `{\"error\":\"Unauthorized: `The JWT is invalid`\"}``",
-                                                               StringComparison.InvariantCultureIgnoreCase) && !redirectThrow)
+                    
+            if (string.IsNullOrEmpty(Token))
             {
-                LogWriteLine($"[DBHandler::Init] Error when connecting to database system! Token invalid!\r\n{e}",
-                             LogType.Error, true);
+                IsEnabled = false;
+                throw new NullReferenceException($"DB_002 {Lang._SettingsPage.Database_Error_EmptyToken}");
             }
-            catch (Exception e) when (!redirectThrow)
+                    
+
+            // Connect to database
+            // Libsql-client-dotnet technically support file based SQLite by pushing `file://` proto in the URL.
+            // But what's the point?
+            _database = await DatabaseClient.Create(opts =>
+                                                    {
+                                                        opts.Url       = Uri;
+                                                        opts.AuthToken = Token;
+                                                    });
+
+            if (_isFirstInit)
             {
-                LogWriteLine($"[DBHandler::Init] Error when (re)initializing database system!\r\n{e}", LogType.Error, true);
+                LogWriteLine("[DbHandler::Init] Initializing database system...");
+                // Ensure table exist at first initialization
+                await
+                    _database
+                       .Execute($"CREATE TABLE IF NOT EXISTS \"uid-{_userIdHash}\" (Id INTEGER PRIMARY KEY AUTOINCREMENT, 'key' TEXT UNIQUE NOT NULL, 'value' TEXT)");
+                _isFirstInit = false;
             }
-            catch (LibsqlException e) when (e.Message.Contains("`api error: `{\"error\":\"Unauthorized: `The JWT is invalid`\"}``",
-                                                               StringComparison.InvariantCultureIgnoreCase))
+            else LogWriteLine("[DbHandler::Init] Reinitializing database system...");
+        }
+        catch (DllNotFoundException e)
+        {
+            LogWriteLine("[DbHandler::Init] Error when connecting to database system! Probably missing Visual C/C++ redist!\r\n" + e,
+                         LogType.Error, true);
+            if (redirectThrow) throw;
+        }
+        // No need to handle all these error catcher with sentry
+        // The error should be handled in the method caller instead
+        catch (LibsqlException e) when (e.Message.Contains("`api error: `{\"error\":\"Unauthorized: `The JWT is invalid`\"}``",
+                                                           StringComparison.InvariantCultureIgnoreCase) && !redirectThrow)
```

Comment:
Just to make sure if the e.Message.Contains is always valid, can we just check the string for ``\"Unauthorized: `The JWT is invalid`\"``?

Reply:
Also, please use ``StringComparison.OrdinalIgnoreCase`` to avoid culture-specific string check.

Reply:
Implemented in https://github.com/CollapseLauncher/Collapse/pull/791/commits/d9de2cac5a3ff89372844e29a4f27df6e7421f42